js/ui/tree_view.d.ts

```diff
@@ -26,6 +30,80 @@ import {
     SearchBoxMixinOptions
 } from './widget/ui.search_box_mixin';
 
+/** @public */
+export type ContentReadyEvent = ComponentEvent<dxTreeView>;
+
+/** @public */
+export type DisposingEvent = ComponentDisposingEvent<dxTreeView>;
+
+/** @public */
+export type InitializedEvent = ComponentInitializedEvent<dxTreeView>;
+
+/** @public */
+export type ItemClickEvent = ComponentNativeEvent<dxTreeView> & {
+    readonly itemData?: any;
+    readonly itemElement?: TElement;
+    readonly itemIndex?: number | any;
```

Comment:
@IgnatovDan, could we make this type more specific? [Documentation says](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#onItemClick) it's `number | object` - does this `object` has a structure, like [dxList item events](https://github.com/DevExpress/DevExtreme/pull/16933/files#diff-6f112a3ec2f1220c5120dec4dd085da62d370c4f15044f761ee28fe915548361R43)?

Reply:
Same for the `ItemContextMenuEvent`

Reply:
Yes, I saw it and wondered why there is 'object' but kept it 'as in docs'.
it seems now it's time to deep into it.

ItemClickEvent/ItemContextMenuEvent:

https://js.devexpress.com/Documentation/21_1/ApiReference/UI_Components/dxTreeView/Configuration/#onItemClick - "itemIndex" is described as "Number | Object"

Generated from declarations: https://github.com/DevExpress/DevExtreme/blob/21_1/js/ui/tree_view.d.ts#L109
Declarations were added here: https://github.com/DevExpress/DevExtreme/pull/5852/files#diff-0ad220537b603d00bc41a3c7617773006bfc16f6c9214778b647f62f7b2c8186R216 - no comments about "why and when an object can be passed in the index member"

https://js.devexpress.com/Documentation/21_1/ApiReference/UI_Components/dxList/Configuration/#onItemClick - similar "Object" but with a clear description WTF.

I have not found a TreeView code that controls this property.
IMHO its value is initialized here: https://github.com/DevExpress/DevExtreme/blob/21_1/js/ui/collection/ui.collection_widget.base.js#L1028

It seems to be an 'integer' (https://api.jquery.com/index/) but I cannot prove it (and, there is a similar function in our 'nonJQ' code).
Also, I even cannot describe 'what does that number means' without research, there are no tests about it (see https://github.com/DevExpress/DevExtreme/blob/21_1/testing/tests/DevExpress.ui.widgets/treeViewParts/events.js).
And I think it's not a good time to research WTF is that "Object" after 'codefreeze': there are no comments about it in initial PR, no tests and it's not easy to review all the possible scenarios and call stacks in code.

Reply:
Ok